addons/mass_mailing_sms/models/mailing_list.py

```diff
@@ -63,3 +63,31 @@ def _mailing_get_opt_out_list_sms(self, mailing):
         opt_out_contacts = subscriptions.filtered(lambda sub: sub.opt_out).mapped('contact_id')
         opt_in_contacts = subscriptions.filtered(lambda sub: not sub.opt_out).mapped('contact_id')
         return list(set(c.id for c in opt_out_contacts if c not in opt_in_contacts))
+
+    def _mailing_list_get_contact_condition(self):
+        email_condition = super(MailingList, self)._mailing_list_get_contact_condition()
+        return "(%s OR %s)" % (
```

Comment:
shouldn't be an AND condition ? (your email shouldn't be in the blacklist and your phone as well)

Reply:
Mm, probably but I'll check the previous behaviour to make sure